lib/mais_orcid_client.rb

```diff
@@ -104,41 +108,32 @@ def last_page?(links)
   end
 
   def get_response(path, allow404: false)
-    response = conn.get("/mais/orcid/v1#{path}")
+    TokenWrapper.refresh(config) do
+      response = conn.get("/mais/orcid/v1#{path}")
 
-    return if allow404 && response.status == 404
+      return if allow404 && response.status == 404
 
-    raise "UIT MAIS ORCID User API returned #{response.status}" if response.status != 200
+      return UnexpectedResponse.call(response) unless response.success?
 
-    body = JSON.parse(response.body).with_indifferent_access
-    raise "UIT MAIS ORCID User API returned an error: #{response.body}" if body.key?(:error)
+      body = JSON.parse(response.body).with_indifferent_access
+      raise UnexpectedResponse::ResponseError, "UIT MAIS ORCID User API returned an error: #{response.body}" if body.key?(:error)
 
-    body
+      body
+    end
   end
 
-  # @return [Faraday::Connection]
   def conn
-    @conn ||= begin
-      conn = Faraday.new(url: base_url) do |faraday|
-        faraday.request :retry, max: 3,
-          interval: 0.5,
-          interval_randomness: 0.5,
-          backoff_factor: 2
-      end
-      conn.options.timeout = 500
-      conn.options.open_timeout = 10
-      conn.headers[:user_agent] = "stanford-library-sul-pub"
-      conn.headers[:authorization] = token
-      conn
+    conn = Faraday.new(url: config.base_url) do |faraday|
+      faraday.request :retry, max: 3,
+        interval: 0.5,
+        interval_randomness: 0.5,
+        backoff_factor: 2
     end
-  end
-
-  def token
-    client = OAuth2::Client.new(client_id, client_secret, site: base_url,
-      token_url: "/api/oauth/token", authorize_url: "/api/oauth/authorize",
-      auth_scheme: :request_body)
-    token = client.client_credentials.get_token
-    "Bearer #{token.token}"
+    conn.options.timeout = 500
+    conn.options.open_timeout = 10
+    conn.headers[:user_agent] = "stanford-library-sul-pub"
+    conn.headers[:authorization] = "Bearer #{config.token}"
+    conn
   end
```

Comment:
Note: We cannot cache the connection object in an accessor in the singleton instance of this class, as it would then continue to persist the expired token.  We open a new connection for each call, to ensure we use whatever the latest token is.